include/IReplySender.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef IREPLYSENDER_HPP
+#define IREPLYSENDER_HPP
+
+#include "IClient.hpp"
+
+namespace api
+{
+    /**
+     * @interface This interface defines the basic behavior of a reply sender
+     * @tparam ResDataType This template parameter refers to the type of the response data type that will be send to the
+     * client.
+     */
+    template <class ResponseDataType>
+    class IReplySender
+    {
+        public:
+            /**
+             * @brief This method sends the response data to the client if the client isn't able to send the data, the
+             * method should throw.
```

Comment:
@brief This method sends the response data to the client. If the client can't send the data, it should throw an exception